app/src/main/java/com/kickstarter/libs/AnalyticEvents.kt

```diff
@@ -853,6 +853,21 @@ class AnalyticEvents(trackingClients: List<TrackingClientType?>) {
         client.track(CTA_CLICKED.eventName, props)
     }
 
+    /**
+     * Sends data associated with the search results page viewed to segment.
+     * @param discoveryParams: DiscoveryParams
+     * @param count: Int
+     * @param sort: DiscoveryParams.Sort
+     */
+    fun trackSearchResultPageViewed(discoveryParams: DiscoveryParams, count: Int, sort: DiscoveryParams.Sort) {
+        val props = AnalyticEventsUtils.discoveryParamsProperties(discoveryParams).toMutableMap()
+        props[CONTEXT_PAGE.contextName] = SEARCH.contextName
+        props[DISCOVER_SORT.contextName] = sort.name.toLowerCase(Locale.ROOT)
+        discoveryParams.term()?.let { props["discover_search_term"] = it }
```

Comment:
Do the specs say that if the search term is null, we don't send the `discover_search_term` property at all? Otherwise we should send an empty string.

Reply:
This is because the getter ``` term() ``` returns an optional

Reply:
I understand that, I'm just trying to figure out if this should be moved into the `discoveryParamsProperties` method in the `analyticEventsUtils` class, or if it should be isolated to this method. Also wondering if we should send `discover_search_term = ""` if the `term()` is empty.

Reply:
```
 .filter(it -> ObjectUtils.isNotNull(it.first.first.term())
                  && StringExt.isPresent(it.first.first.term())
                  && it.first.first.sort() != defaultSort
                  && IntegerUtils.intValueOrZero(it.second) == 1)
```

this ensures ``` term() ``` is never empty

Reply:
Offline chat: these two properties are unique to this event, and therefore shouldn't be moved to the `discoverProperties()` function in the `AnalyticEventsUtils` class.